design/secrets_provider_milestone1_2_design.md

```diff
@@ -0,0 +1,367 @@
+
+1. Configure K8s Secrets with ***conjur-map\*** metadata defined 
+
+   - *Optional:* add label to K8s Secret ***(Milestone 1)***
+
+2. Create Service Account for Secrets Provider 
+
+3. Create Role / ClusterRole for the Service Accounts with **get**/**update** privileges
+
+4. Create RoleBinding / ClusterRoleBinding and bind the Service Accounts to the role from previous step
+
+5. Create/Deploy Secrets Providers as Jobs in deployment manifest(s) ***(Milestone 1)***
+
+6. - If label filtering is defined, add "K8S_SECRET_LABEL" in the Secrets Provider manifest(s) ***(Milestone 1)***
+
+
+
+#### Customer Experience
+
+In order to preserve restrictions on Conjur secrets, a customer can decide to either:
+
+1. Deploy 1 Secrets Provider per namespace, giving the Secrets Provider access to all secrets
+2. Deploy 1 Secrets Provider per app / (group of apps), giving the Secrets Provider only access to the secrets they have been given access to in Conjur. Should the user require further separation of duties on Conjur secrets, they will need to deploy another Secrets Provider
+
+The Secrets Provider host will resemble the following:
+
+```yaml
+- !host
+  id: secrets-provider-1
+  annotations:
+    authn-k8s/namespace: prod-namespace
+    authn-k8s/service-account: prod-sa
+    authenticator-container-name: cyberark-secrets-provider
+ 
+- !permit
+  resources: !variable super_secret
+  role: !layer secrets
+  privileges: [ read, execute ]
+ 
+- !grant
+  role: !layer secrets
+  members:
+    - !host conjur/authn-k8s/secrets-provider-1
+```
+
+The Secrets Provider manifest will be defined as the following:
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: secrets-provider-job
+spec:
+  template:
+    spec:
+      serviceAccountName: prod-sa
+      containers:
+      - image: secrets-provider:latest
+        name: cyberark-secrets-provider-1
+....
+```
+
+
+
+#### Code changes
+
+1. Support Job application identity 
+2. Enhance Batch retrieval
+
+*Support Job application identity*
+
+To authenticate to Conjur, the Secrets Provider application identity (or Resource Restrictions) will be the characteristics of the Pod that it is running in. Currently, we do not have a "Job" application identity granularity so to allow the Secrets Provider the ability authenticate to Conjur, the customer will need to choose between the Namespace or Service Account granularities. If there is a requirement to support a "Job" granularity this can be done by adding code to the authenticator in Conjur.
+
+*Enhance Batch retrieval*
+
+At current, the Secrets Provider code looks at the "K8S_SECRET" environment variable in each of the Pods and preforms a batch retrieval against the Conjur API to get the relevant Conjur Secrets. If there is a failure in retrieving one of the secrets, the request fails and no secrets are returned to the request Pod.
```

Comment:
I think that Inbal wanted to immitate the way summon works to retrieve secrets, and there is already existing input format for it that differs from what you say